src/Orleans.Runtime/Core/InsideRuntimeClient.cs

```diff
@@ -449,21 +452,30 @@ public void SniffIncomingMessage(Message message)
             catch (Exception exc2)
             {
                 logger.Warn(ErrorCode.Runtime_Error_100329, "Exception during Invoke of message: " + message, exc2);
-                if (message.Direction != Message.Directions.OneWay)
-                    SafeSendExceptionResponse(message, exc2);
 
-                if (exc2 is OrleansTransactionInDoubtException)
+                try
                 {
-                    this.logger.LogError(exc2, "Transaction failed due to in doubt transaction");
+                    if (exc2 is OrleansTransactionInDoubtException)
+                    {
+                        this.logger.LogError(exc2, "Transaction failed due to in doubt transaction");
+                    }
+                    else if (TransactionContext.GetTransactionInfo() != null)
+                    {
+                        // Must abort the transaction on exceptions
+                        TransactionContext.GetTransactionInfo().IsAborted = true;
+                        var abortException = (exc2 as OrleansTransactionAbortedException) ??
+                            new OrleansTransactionAbortedException(TransactionContext.GetTransactionInfo().TransactionId, exc2);
+                        this.transactionAgent.Value.Abort(TransactionContext.GetTransactionInfo(), abortException);
+                    }
                 }
-                else if (TransactionContext.GetTransactionInfo() != null)
+                catch (Exception) { }
```

Comment:
I don't quite understand why we are ignoring exception here

Reply:
if it is not important, should we at least log it?

Reply:
Squeshed because we're already in error handling logic and there is nothing that can be done, but yeah, probably best not to squash these.  Fixed